akka-docs-new/src/main/paradox/guide/tutorial_1.md

```diff
@@ -1,41 +1,235 @@
+This ordering is strict, all `postStop()` hooks of the children are called before the `postStop()` hook of the parent
+is called.
+ 
+The family of these lifecycle hooks is rich, and we recommend reading the actor lifecycle (TODO: reference-doc) section
+of the reference for all the details.
+
+### Hierarchy and failure handling (Supervision)
+
+Parents and children are not only connected by their lifecycles. Whenever an actor fails (throws an exception or
+an unhandled exception bubbles out from `receive`) it is temporarily suspended. The failure information is propagated
+then to the parent, which can decide what to do with the child actor now. The default _supervisor strategy_ is to
+stop and restart the child. If you don't change the default strategy all failures result in a restart. We won't change
+the default for now in this simple experiment:
+
+@@snip [Hello.scala](../../../test/scala/tutorial_1/ActorHierarchyExperiments.scala) { #supervise }
+
+After running the snippet, we see the following output on the console:
+
+```
+supervised actor started
+supervised actor fails now
+supervised actor stopped
+supervised actor started
+[ERROR] [03/29/2017 10:47:14.150] [testSystem-akka.actor.default-dispatcher-2] [akka://testSystem/user/supervising-actor/supervised-actor] I failed!
+java.lang.Exception: I failed!
+        at tutorial_1.SupervisedActor$$anonfun$receive$4.applyOrElse(ActorHierarchyExperiments.scala:57)
+        at akka.actor.Actor$class.aroundReceive(Actor.scala:513)
+        at tutorial_1.SupervisedActor.aroundReceive(ActorHierarchyExperiments.scala:47)
+        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:519)
+        at akka.actor.ActorCell.invoke(ActorCell.scala:488)
+        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
+        at akka.dispatch.Mailbox.run(Mailbox.scala:224)
+        at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
+        at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
+        at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
+        at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
+        at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
+```
+
+We see that after failure the actor is stopped and immediately started again. We also see a log entry reporting the
+exception that was handled, in this case our test exception. We only used here `preStart()` and `postStop()` hooks
+which are the default to be called after and before restarts, so we cannot distinguish from inside the actor if it
+was started for the first time or restarted. This is usually the right thing to do, the purpose of the restart is to
+set the actor in a known-good state, which usually means a clean starting stage. What actually happens though is
+that *the preRestart()` and `postRestart()` methods are called which, if not overridden, by default delegate to
```

Comment:
I think it's also important to mention that a new instance of the actor is created when it's restarted, and thereby it's state is "reset".